### PR TITLE
Added InspectCode Support

### PR DESCRIPTION
**Pull request by yvschmid**
InspectCode support #334

### CODE REVIEW

File: src/Cake.Common/Tools/InspectCode/InspectCodeAliases.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+using Cake.Core;
+using Cake.Core.Annotations;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.InspectCode
+{
+    /// <summary>
+    /// Contains functionality related to Resharper's code inspection.
+    /// </summary>
+    [CakeAliasCategory("InspectCode")]
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/InspectCode/InspectCodeAliases.cs]: Maybe this should be `ReSharper`? Would look better in the documentation.

Comment (gep13) [inline src/Cake.Common/Tools/InspectCode/InspectCodeAliases.cs]: That would make sense if/when DupFinder is added, as this is also a ReSharper Tool.

**Comment by patriksvensson** [comment]
@yvschmid Thanks for this! :+1: 

I've added a comment about a very minor thing.

File: src/Cake.Common/Tools/InspectCode/InspectCodeAliases.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public static class InspectCodeAliases
+    {
+        /// <summary>
+        /// Analyses the specified solution with Resharper's InspectCode.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="solution">The solution.</param>
+        [CakeMethodAlias]
+        public static void InspectCode(this ICakeContext context, FilePath solution)
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/InspectCode/InspectCodeAliases.cs]: Could add a `[CakeAliasCategory("InspectCode")]` here.

File: src/Cake.Common/Tools/InspectCode/InspectCodeAliases.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        /// <summary>
+        /// Analyses the specified solution with Resharper's InspectCode,
+        /// using the specified settings.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="solution">The solution.</param>
+        /// <param name="settings">The settings.</param>
+        [CakeMethodAlias]
+        public static void InspectCode(this ICakeContext context, FilePath solution, InspectCodeSettings settings)
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/InspectCode/InspectCodeAliases.cs]: Could add a `[CakeAliasCategory("InspectCode")]` here.

File: src/Cake.Common/Tools/InspectCode/InspectCodeAliases.cs (hunk 1 above)
Comment (patriksvensson) [inline src/Cake.Common/Tools/InspectCode/InspectCodeAliases.cs]: Yes, These methods would be grouped like ReSharper/InspectCode which would be very nice.

**Comment by yvschmid** [comment]
So do you mean like this or should we move the whole thing to a `Tools.ReSharper.InspectCode` namespace?

**Comment by patriksvensson** [comment]
What you've done looks good to me. We can move it to a ReSharper namespace once we got more tools. Perfect :+1:

**Comment by patriksvensson** [comment]
@yvschmid Looks good for a merge. Would be nice if you could rebase this to a single commit.

Great work :+1:

**Comment by patriksvensson** [comment]
Implements #334

**Comment by patriksvensson** [comment]
Any reason for closing this PR? :/

**Comment by yvschmid** [comment]
Sorry, i was messing arround with the history of my fork and had to learn the hard way that you can't reopen PR after you closed them by force pushing.

New PR is here #355

**Comment by patriksvensson** [comment]
Ah, I understand. Stuff like that happen :)

**Pull request closed without merging**